src/main/java/com/odysseusinc/arachne/portal/service/impl/BaseStudyServiceImpl.java

```diff
@@ -1021,23 +1021,25 @@ public T findByIdInAnyTenant(final Long studyId) {
         return studyRepository.findByIdInAnyTenant(studyId);
     }
 
-    @PostAuthorize("@ArachnePermissionEvaluator.addPermissions(principal, returnObject )")
     @Override
-    public T findWorkspaceForUser(Long userId) throws NotExistException {
+    public T findWorkspaceForUser(IUser user, Long userId) throws NotExistException {
 
         T workspace = studyRepository.findWorkspaceForUser(userId);
+        
+        workspace = (T)this.getStudy(user, workspace.getId()).getStudy();
+        
```

Comment:
Looks not enough clean to me

Reply:
getStudy returns WorkspaceUser object and you cant get study without cast

Reply:
probably I missed smth, just to clarify
User and Workspace has 1-to-1 relation, right?
why userId is not sufficient to find workspace if one exits?